report/sections/digital-twins.tex

```diff
@@ -3,48 +3,345 @@ \chapter{Digital Twins}
+
+Finally, it is useful to have early warnings in case critical situations occur, such as a pH value that is too high (or too low), or fluctuations in
+temperature.
+In this regard, provision has been made for the digital twin to propagate messages so that timely action can be taken.
+
+The use of a digital twin in this scenario is advantageous for several reasons: human intervention to perform milk analysis is minimized as there is
+real-time monitoring of all key parameters, thus reducing the time to go into production.
+As a consequence of real-time analysis, there is a guarantee that the milk used reflects all requirements thus maximizing product quality and yield.
+Also as a consequence of the real-time analysis of all parameters, an alarm system is available to enable timely action to avoid waste and loss of time.
+
 \subsubsection{Thing Model}
-Properties: Manufacturer, Serial number, pH, minPH, maxPH, temperature, minTemp, maxTemp, quantityOfmilk
-Events: temperatureOutOfRange, pHoutOfRange
-Actions: Open Valve
 
-\section{Digital twins for anticipazione guasti}
+\lstinputlisting[label={lst:milk-tank-tm},caption={Milk tank Thing Model.}]{lst/milk-tank.tm.jsonld}
+
+As you can see from the listing~\ref{lst:milk-tank-tm}, two static properties were defined: \textit{serialNo} and \textit{Manufacturer}.
+Those two properties represent the serial number of the tank and the manufacturer, respectively.
+Since those properties come with the tank, they are read-only.
+
+Another read-only property is \textit{availableMilk}, which represents the level of milk in the tank. Since the level is calculated directly by the
+tank, it is read-only so that no external operation could alter the value. This property is used directly by the \textit{milk-planning} to
+plan the reordering of milk.
+
+The only available action is \textit{openValve}, which is used to open the valve and start the production process. The action is represented by a boolean value representing the current state of the valve.
+
+This particular digital twin is characterized by the fact that is linked by other two devices: \textit{temperature-sensor} and \textit{ph-sensor}.
+Those two sensors are independent devices that are attached to the tank and are used to monitor the temperature and the pH of the milk.
+In the \textbf{Thing Model} specification the connection is represented by the \texttt{links} construct showing that the ``root thing'' is connected
+with other ``things''.
+This kind of connection enables semantic information about how things are connected and which kinds of relationships occur between them.
+
+The \textit{temperature} sensor defines the following properties: \textit{temperature}, \textit{min-temp} and \textit{max-temp}. The former represents the current temperature from the sensor; the last two properties are useful to define the range of temperature that is considered normal.

```

Comment:
Say something about the fact that temperature is a submodule and it is reused as a parameter also for the packaging machine